src/modules/fancyzones/lib/GenericKeyHook.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <functional>
+#include "pch.h"
+
+template<int... keys>
+class GenericKeyHook
+{
+public:
+
+    GenericKeyHook(std::function<void(bool)> extCallback)
+    {
+        callback = std::move(extCallback);
```

Comment:
```suggestion
      if (!callback)
      {
          callback = std::move(extCallback);
      }
      else
      {
          throw std::runtime_error("Only one instance of a certain callback is supported!");
      }
```

Reply:
Unfortunately, this would throw when turning FZ off and then on again.

Reply:
Clearing `callback` in `~GenericKeyHook` seems to fix the crash but I couldn't find a guarantee that it will always behave like this. Basically we have to rely on `winrt::com_ptr` calling the destructor for `FancyZones`.